apps/namadillo/src/hooks/useTransactionCallbacks.tsx

```diff
@@ -19,4 +21,27 @@ export const useTransactionCallback = (): void => {
   useTransactionEventListener("Bond.Success", onBalanceUpdate);
   useTransactionEventListener("Unbond.Success", onBalanceUpdate);
   useTransactionEventListener("Withdraw.Success", onBalanceUpdate);
+
+  const store = createStore();
+
+  useTransactionEventListener("VoteProposal.Success", (args) => {
+    // TODO: refactor this after event subscription is enabled on indexer
+    shouldUpdateProposal(true);
+
+    // [0] is fine for time being as we can only vote for one proposal at a time
+    const proposalId = args.detail.data[0].proposalId;
+    const { refetch: refetchProposalFamily } = store.get(
+      proposalFamily(proposalId)
+    );
+    refetchProposalFamily();
```

Comment:
Just to double check: Is this needed because setting `refetchInterval` on the atom will mean we need to wait 1 second before the first refetch?

Reply:
Hmm I kind of copied it mindlessly from the example above :D but I think you are right. I think in this case we can just remove it, I think there is 0% chance we get proposal status right away. I will double check this